test/unit/codegen/codegen_cpp_visitor.cpp

```diff
@@ -422,3 +422,34 @@ SCENARIO("Check code generation for TABLE statements", "[codegen][array_variable
         }
     }
 }
+
+SCENARIO("Check codegen for MUTEX and PROTECT", "[codegen][mutex_protect]") {
+    GIVEN("A mod file containing MUTEX & PROTECT") {
+        std::string const nmodl_text = R"(
+            NEURON {
+                SUFFIX TEST
+                RANGE tmp
+            }
+            PARAMETER {
+                tmp = 10
+            }
+            INITIAL {
+                MUTEXLOCK
+                tmp = 11
+                MUTEXUNLOCK
+                PROTECT tmp = 12
+            }
+        )";
+
+        THEN("Code with OpenMP critical sections is generated") {
+            auto const generated = get_cpp_code(nmodl_text);
+            std::string expected_code = R"(#pragma omp critical TEST {
```

Comment:
@alkino : we missed this in review! the [valid syntax](https://www.openmp.org/spec-html/5.0/openmpsu89.html) for named critical section is : `#pragma omp critical (name)` i.e. `( )` are missing and hence it's compilation error!

Reply:
oupsy

Reply:
is there no test for that?

Reply:
We should have added the keyword in one of the mod files here: https://github.com/BlueBrain/nmodl/tree/master/test/integration/mod that get compiled in the CI